### PR TITLE
Fix Xunit runner transparent proxy error problem on net452 target

### PR DESCRIPTION
**Pull request by Arkatufus**
Fix attempt for #2881

### CODE REVIEW

File: src/core/Akka.MultiNodeTestRunner.Shared.Tests/MultiNodeTestRunnerDiscovery/DiscoverySpec.cs [hunk 1]
```diff
@@ -46,7 +46,7 @@ public void Public_props_and_fields_are_considered()
 
         private static Dictionary<string, List<NodeTest>> DiscoverSpecs()
         {
-            using (var controller = new XunitFrontController(AppDomainSupport.IfAvailable, new System.Uri(Assembly.GetExecutingAssembly().CodeBase).LocalPath))
+            using (var controller = new XunitFrontController(AppDomainSupport.IfAvailable, new System.Uri(typeof(DiscoveryCases).GetTypeInfo().Assembly.CodeBase).LocalPath))
```
Comment (Aaronontheweb) [inline src/core/Akka.MultiNodeTestRunner.Shared.Tests/MultiNodeTestRunnerDiscovery/DiscoverySpec.cs]: If this passes in this PR, I'll revert my changes on https://github.com/akkadotnet/akka.net/pull/2875

File: src/core/Akka.MultiNodeTestRunner.Shared.Tests/NodeMessageHelpers.cs [hunk 2]
```diff
@@ -20,12 +20,13 @@ namespace Akka.MultiNodeTestRunner.Shared.Tests
...
     {
         internal const string DummyRoleFor = "Dummy_role_for_";
+        internal static readonly Random Random = new Random();
 
         public static IList<NodeTest> BuildNodeTests(IEnumerable<int> nodeIndicies)
         {
-            var methodName = Faker.Generators.Strings.GenerateAlphaNumericString();
-            var className = Faker.Generators.Strings.GenerateAlphaNumericString();
-            var testName = Faker.Generators.Strings.GenerateAlphaNumericString();
+            var methodName = AlphaNumericString();
```
Comment (Aaronontheweb) [inline src/core/Akka.MultiNodeTestRunner.Shared.Tests/NodeMessageHelpers.cs]: Good

File: src/core/Akka.MultiNodeTestRunner.Shared.Tests/TestRunShutdownSpec.cs [hunk 3]
```diff
@@ -40,6 +41,8 @@ public void TestCoordinatorEnabledMessageSink_should_receive_TestRunTree_when_En
             foreach (var message in allMessages)
                 consoleMessageSink.Tell(message);
 
+            Task.Delay(1000).Wait();
```
Comment (Aaronontheweb) [inline src/core/Akka.MultiNodeTestRunner.Shared.Tests/TestRunShutdownSpec.cs]: Technically, shouldn't need this - I was able to get the spec to pass without it on https://github.com/akkadotnet/akka.net/pull/2875

File: src/core/Akka.MultiNodeTestRunner.Shared.Tests/TestRunShutdownSpec.cs [hunk 4]
```diff
@@ -22,7 +23,7 @@ public class TestRunShutdownSpec : AkkaSpec
         [Fact]
         public void TestCoordinatorEnabledMessageSink_should_receive_TestRunTree_when_EndTestRun_is_received()
         {
-            var consoleMessageSink = Sys.ActorOf(Props.Create(() => new ConsoleMessageSinkActor(false)));
+            var consoleMessageSink = Sys.ActorOf(Props.Create(() => new ConsoleMessageSinkActor(true)));
```
Comment (Aaronontheweb) [inline src/core/Akka.MultiNodeTestRunner.Shared.Tests/TestRunShutdownSpec.cs]: This is correct

File: src/core/Akka.MultiNodeTestRunner.Shared/Discovery.cs [hunk 5]
```diff
@@ -17,9 +17,13 @@
 using Xunit.Abstractions;
 using Xunit.Sdk;
 
-namespace Akka.MultiNodeTestRunner
+namespace Akka.MultiNodeTestRunner.Shared
 {
-    internal class Discovery : MarshalByRefObject, IMessageSink, IDisposable
+#if CORECLR
```
Comment (Aaronontheweb) [inline src/core/Akka.MultiNodeTestRunner.Shared/Discovery.cs]: nice catch

Comment (Arkatufus) [inline src/core/Akka.MultiNodeTestRunner.Shared/Discovery.cs]: Actually, that was a mistake, the `NetCore` part was from the other PR to add core support, I merely stripped all the core support and re submitted it.

**Pull request merged**